### ISSUE TITLE
ALVR+SteamVR broken on latest unstable

### ISSUE THREAD

**Issue by eyJhb** [NixOS/nixpkgs#352304]
### Describe the bug
Using ALVR + SteamVR on latest unstable will not work, and it will crash as soon as a headset is connected.
This is likely introduced here https://github.com/NixOS/nixpkgs/pull/341219 , and there is a fix here https://github.com/NixOS/nixpkgs/pull/351928 (this cannot be cherry picked, there are more things in staging-next that is required for this to work according to k900).
Running that PR has worked for me without any issues. 

It is also possible to run revision fb4afaa052082fe0d9f5a325047deb429e782bdf before the decimate of Steam.

There  is also another PR trying to "un-decimate" Steam here https://github.com/NixOS/nixpkgs/pull/352004 .

This issue can be closed once either https://github.com/NixOS/nixpkgs/pull/351928 or https://github.com/NixOS/nixpkgs/pull/352004 has been merged into unstable latest, and confirmed to be working in there.

### Steps To Reproduce
1. Open Steam
2. Open ALVR
3. Preass Launch  SteamVR in  ALVR
4. Connect headset
5. Crash

### Expected behavior
That it works, and I can see  SteamVR in my headset.

This issue is made to move the discussion away from the PR here https://github.com/NixOS/nixpkgs/pull/308097

**Comment by LuNeder** [comment]
> there is a fix here #351928 (this cannot be cherry picked, there are more things in staging-next that is required for this to work according to k900). Running that PR has worked for me without any issues.
> 
> It is also possible to run revision [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf) before the decimate of Steam.
> 
> There is also another PR trying to "un-decimate" Steam here #352004 .
> 
> This issue can be closed once either #351928 or #352004 has been merged into unstable latest, and confirmed to be working in there.


Hey! Can you confirm that it actually still works for you on #352004 or #351928 on their latest commits/force-pushes? Switching my system to those don’t fix SteamVR/ALVR at all for me!

Aditionally: while [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf) does work, it does not seem to be the parent commit of #341219 (git is weird). If I get the parent of [the first commit](https://github.com/NixOS/nixpkgs/commit/7014f869474850e8e8a43db61aa8566215f9a1a6) in that PR (which is [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6) according to github, it seems to have been merged into unstable before fb4afaa - but its parents are way more recent than fb4afaa), SteamVR is already broken there so it feels weird for the decimate pr to be the cause. If my observations are correct, I imagine that the culprit for the breakage should be somewhere between 273673e’s parent and fb4afaa’s parent commits (but I have no idea how to check those commits if github history page isn’t reliable enough for that).

**Comment by Atemu** [comment]
> (ALVR/SteamVR continues broken on [undecimate](https://github.com/LuNeder/nixos-config/commit/6a78143a684622bc6de314bdbfd8d592662102f7#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0) and on the [new K900 PR](https://github.com/LuNeder/nixos-config/commit/b4174da1e1bfc3d0eb73d672904fa7f65e3c71ef#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0) exactly the same way as on latest unstable)

Could you confirm that it works on the commit before https://github.com/NixOS/nixpkgs/pull/341219 and is broken after?

This should work: https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6
This should break: https://github.com/NixOS/nixpkgs/commit/a3ccb7f562a93826ad0112adf20dc7f697b7713e

**Comment by LuNeder** [comment]
> Could you confirm that it works on the commit before #341219 and is broken after?
> 
> This should work: [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6) This should break: [a3ccb7f](https://github.com/NixOS/nixpkgs/commit/a3ccb7f562a93826ad0112adf20dc7f697b7713e)

No. As I mentioned on the above comment, it is broken on a3ccb7f but also broken on 273673e. It works fine on fb4afaa, which on nixos-unstable commit history is immediately before the decimate pr commits, but while fb4afaa was merged after 273673e into nixos-unstable fb4afaa’s parents are older than 273673e ones. Due to that I assume that the ALVR breaking culprit is not actually the decimate PR, but some commit between 273673e’s parent and fb4afaa’s parent commit.

As I mentioned on my above comment:


> while [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf) does [work](https://github.com/LuNeder/nixos-config/commit/3d348fd70436164de29911bd2e69436e565d1821#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0), it does not seem to be the parent commit of #341219 (git is weird). If I get the parent of [the first commit](https://github.com/NixOS/nixpkgs/commit/7014f869474850e8e8a43db61aa8566215f9a1a6) in that PR (which is [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6) according to github, it seems to have been merged into unstable before [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf) - but its parents are way more recent than [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf)), SteamVR is already broken there so it feels weird for the decimate pr to be the cause. If my observations are correct, I imagine that the culprit for the breakage should be somewhere between [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6)’s parent and [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf)’s parent commits (but I have no idea how to check those commits if github history page isn’t reliable enough for that).

**Comment by Atemu** [comment]
@eyJhb does https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6 also not work for you?

**Comment by LuNeder** [comment]
Just retested it to make sure, and that's still true for me: 
- Working on [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf)
- Already broken on [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6)
- Still broken on #351928 

Unsure how helpful this is, but here's a result of a store diff between fb4afaa (newest known working for now) and 273673e (oldest known broken for now): [alvr-gen-diff4.txt](https://github.com/user-attachments/files/17573771/alvr-gen-diff4.txt)

I was talking with @eyJhb on the nix gaming room on Matrix, and we have no idea how to do the bisect thing to figure out which commits exist between [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6)’s parent and [fb4afaa](https://github.com/NixOS/nixpkgs/commit/fb4afaa052082fe0d9f5a325047deb429e782bdf)’s parent so we could try those and know exactly which commit broke ALVR. If there's a way of listing that on GitHub or something, or if someone better at git could try doing the bisect, that'd be very helpful

**Comment by eyJhb** [comment]
> @eyJhb does [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6) also not work for you?

Yes, this work for me. Full rebuild, reboot, and I can get the SteamVR inside my headset.
I've also double checked https://github.com/NixOS/nixpkgs/pull/351928 , full rebuild, full reboot, and it works inside SteamVR.

AFAIK all the times I've ran the bisect wit h`--first-parent`, I end up at that merge, and I haven't dug further.

**Comment by Atemu** [comment]
So it appears @eyJhb's case is different from @LuNeder and this is indeed fixed by https://github.com/NixOS/nixpkgs/pull/351928.

@LuNeder please create a separate issue and do this bisect: https://github.com/NixOS/nixpkgs/issues/352304#issuecomment-2447708058

**Comment by eyJhb** [comment]
> > (ALVR/SteamVR continues broken on [undecimate](https://github.com/LuNeder/nixos-config/commit/6a78143a684622bc6de314bdbfd8d592662102f7#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0) and on the [new K900 PR](https://github.com/LuNeder/nixos-config/commit/b4174da1e1bfc3d0eb73d672904fa7f65e3c71ef#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0) exactly the same way as on latest unstable)
> 
> Could you confirm that it works on the commit before #341219 and is broken after?
> 
> This should work: [273673e](https://github.com/NixOS/nixpkgs/commit/273673e839189c26130d48993d849a84199523e6) This should break: [a3ccb7f](https://github.com/NixOS/nixpkgs/commit/a3ccb7f562a93826ad0112adf20dc7f697b7713e)

Also confirming that this statement is correct for me as well.

```
eyjhb@chronos ~/p/n/nixpkgs (master)> git bisect start --first-parent
status: waiting for both good and bad commits
eyjhb@chronos ~/p/n/nixpkgs (master|BISECTING)> git bisect good 273673e
status: waiting for bad commit, 1 good commit known
eyjhb@chronos ~/p/n/nixpkgs (master|BISECTING)> git bisect bad a3ccb7f
Bisecting: 0 revisions left to test after this (roughly 1 step)
[70cb669a2f5608583caedde7e691b104f83ad718] buildFHSEnv: fix nested fhsenvs with LD_PRELOAD
eyjhb@chronos ~/p/n/nixpkgs ((70cb669a)|BISECTING)> git bisect good
a3ccb7f562a93826ad0112adf20dc7f697b7713e is the first bad commit
commit a3ccb7f562a93826ad0112adf20dc7f697b7713e
Author: K900 <me@0upti.me>
Date:   Thu Sep 19 18:21:35 2024 +0300

    steam: decimate

 nixos/modules/programs/steam.nix                           |   2 +-
 pkgs/build-support/build-fhsenv-bubblewrap/buildFHSEnv.nix |   3 +
 pkgs/games/steam/default.nix                               |  17 +--
 pkgs/games/steam/fhsenv.nix                                | 431 ++++++++++++++++--------------------------------------------
 pkgs/games/steam/runtime-wrapped.nix                       |  45 -------
 pkgs/games/steam/runtime.nix                               |  37 ------
 pkgs/top-level/aliases.nix                                 |   1 +
 pkgs/top-level/all-packages.nix                            |   1 -
 8 files changed, 118 insertions(+), 419 deletions(-)
 delete mode 100644 pkgs/games/steam/runtime-wrapped.nix
 delete mode 100644 pkgs/games/steam/runtime.nix
```

(all tested, all fully rebooted)

**Comment by eyJhb** [comment]
This has very much been fixed AFAIK.
